xc/xenops_server_xen.ml

```diff
@@ -683,6 +683,28 @@ let upgrade_for_migration ~xc features =
    with Xenctrlext.Unix_error(Unix.ENOSYS, _) ->
      debug "xc_get_msr_arch_caps: ENOSYS"
 
+let i2b i =
+  match Base64.encode (Int64.to_string i) with
+    Ok x -> x
+  | _ -> failwith ("Failed to Base64 Encode: " ^ Int64.to_string i)
+
+let serialise policy =
+  Array.fold_left (fun s acc ->
+      match s with
+        "" -> acc
+      | _ -> match acc with
+          "" -> s
+        | _ -> s ^ "," ^ acc
```

Comment:
it would be more efficient to use some sort of buffer

Reply:
Use the `Buffer` module to incrementally build a large string.

Reply:
It looks like this takes an array of `int64` and wants to serialise it. Is it worth doing the Base64 encoding? Why not simply:
```
a |> Array.map Int64.to_string |> Array.to_list |> String.concat ","
```

How many entries are in these arrays?